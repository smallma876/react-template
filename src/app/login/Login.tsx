import React, { lazy, Suspense } from 'react';
import { Route, Routes } from 'react-router-dom';

const Home = lazy(() => import('./pages/home/Home'));

const Login: React.FC = () => {
  return (
    <Suspense>
      <Routes>
        <Route path="/" element={<Home />} />
      </Routes>
    </Suspense>
  );
};

export default Login;
