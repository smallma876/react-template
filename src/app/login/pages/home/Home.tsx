import React from 'react';

const Home: React.FC = () => {
  return (
    <div>
      <div>OAUTH 2.0</div>
      <div>Login with: </div>
      <div>
        <ul>
          <li>
            <a href="#">Google</a>
          </li>
          <li>
            <a href="#">Faceebook</a>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Home;
