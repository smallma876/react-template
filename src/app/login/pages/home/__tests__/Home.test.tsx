import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Login from '../../../Login';
import Home from '../Home';

describe('home test', () => {
  it('render home', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Login />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
