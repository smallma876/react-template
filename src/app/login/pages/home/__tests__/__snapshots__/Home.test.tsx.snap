// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home test render home 1`] = `<div />`;
